src/modules/Builder/features/ActivitySettings/SubscalesConfiguration/SubscaleLookupTable/SubscaleLookupTable.hooks.ts

```diff
@@ -0,0 +1,53 @@
+import { useEffect, useState } from 'react';
+import { v4 as uuidv4 } from 'uuid';
+
+import { DataTableItem, ImportedFile } from 'shared/components';
+
+import { ModalType, Steps, SubscaleLookupTableSetupHookProps } from './SubscaleLookupTable.types';
+
+export const useSubscaleLookupTableSetup = ({ tableData }: SubscaleLookupTableSetupHookProps) => {
+  const [modalType, setModalType] = useState<ModalType | null>(null);
+  const [step, setStep] = useState<Steps>(0);
+  const [data, setData] = useState<DataTableItem[]>();
+  const [error, setError] = useState<JSX.Element | null>(null);
+
+  const onFileReady = (file: ImportedFile | null) => {
+    const mappedData = (file?.data ?? []).map((item) => ({
+      ...item,
+      id: uuidv4(),
+    }));
+    setData(mappedData);
```

Comment:
Do y need to set setData([]) in some cases?

Reply:
in some cases it's possible, it depends on provided data from csv file. What is your idea?

Reply:
const mappedData = file?.data?.map((item) =>

setData(mappedData ?? [ ]);